src/compression.rs

```diff
@@ -1,13 +1,19 @@
 use flate2::read::GzDecoder;
-use std::io::Read;
+use flate2::write::GzEncoder;
+use flate2::Compression;
+use std::io::{Read, Write};
 
 pub trait Decompress {
```

Comment:
We could rename the trait to `Compression` to make it clearer what it is for.

Reply:
Maybe `CompressionAlgorithm`?